source/Sylvan.Data.Csv/CsvDataWriter+Sync.cs

```diff
@@ -14,16 +15,29 @@ public long Write(DbDataReader reader)
 	{
 		var c = reader.FieldCount;
 		var fieldInfos = new FieldInfo[c];
-		var schema = (reader as IDbColumnSchemaGenerator)?.GetColumnSchema();
+		ReadOnlyCollection<DbColumn>? schema = null;
+		try
+		{
+			// on .NET 48, this will throw NSE
+			// on that runtime, we'll treat it as null.
+			schema = reader.GetColumnSchema();
+		}
+		catch (NotSupportedException)
+		{
+			schema = null;
+		}
```

Comment:
Catching `NotSupportedException` is a bit of a smell, and we do have a method specifically to check this (which implements exactly the precondition that causes `GetColumnSchema` to throw). 

```suggestion
		ReadOnlyCollection<DbColumn>? schema = reader.CanGetColumnSchema() ? reader.GetColumnSchema() : null;
```